### PR TITLE
fix(AWS) When using aws_profile, grab temporary credentials from the session.

### PR DESCRIPTION
**Pull request by Jiafi**
…Mostly used for SSO and temporary credentials

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py
Comment (fahadkh) [inline metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py:95]: This code is fairly duplicative of the `get_session` code above. I wonder if it's worth refactoring both `get_session` and `get_credentials` to use this common `aws` profile details extraction logic.

Comment (Jiafi) [reply metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py:95]: how about now?

Comment (fahadkh) [reply metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py:95]: Makes sense to use the credentials of the session to me.

File: metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py
Comment (fahadkh) [inline metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py:232]: Assuming we want to continue using boto3 as directly as possible, we could directly call out to `self.source_config.aws_config.get_session().get_credentials()` here

Comment (Jiafi) [reply metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py:232]: Gonna leave that up to a datahub maintainer to decide. There is a slight bit of logic in the `get_credentials()` function for `None` Checking

**Comment by Jiafi** [comment]
@aditya-radhakrishnan fixed the typing bug.  That seemed to be the only thing failing.

File: metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py
Comment (aditya-radhakrishnan) [reply metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py:232]: I think the current pattern is fine!

**Review by aditya-radhakrishnan — APPROVED** [review]
Overall this LGTM! Have you tested this out locally at all?

**Comment by Jiafi** [comment]
> Overall this LGTM! Have you tested this out locally at all?

It has been tested locally and works!

**Review by treff7es — APPROVED** [review]

**Comment by treff7es** [comment]
@Jiafi Thanks for the contribution!

**Pull request merged**
